### PR TITLE
All Platforms: Create "Manage features" Sample

### PR DESCRIPTION
**Pull request by amweis226**
# Description

Summary of the change and any relevant info.

## Type of change

- New sample implementation

## Platforms tested on

- [x] WPF .NET 6
- [x] WPF Framework
- [x] WinUI
- [x] MAUI WinUI
- [x] MAUI Android
- [x] MAUI iOS
- [x] MAUI MacCatalyst

## Checklist

- [x] Self-review of changes
- [x] All changes work as expected on all affected platforms
- [x] There are no warnings related to changes
- [x] Code is commented and follows .NET conventions and standards
- [x] Codemaid and XAML styler extensions have been run on every changed file
- [x] No unrelated changes have been made to any other code or project files
- [x] Screenshots are correct size and display in description tab (800 x 600 on Windows, 600 height mobile screenshots for MAUI)

### CODE REVIEW

File: src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+            <StackPanel>
+                <Label Content="Select an option." />
+                <ComboBox x:Name="MethodPicker"
+                          IsEditable="False"
+                          IsEnabled="True"
+                          SelectionChanged="OnComboBoxSelectionChanged" />
+                <Label x:Name="InstructionLabel" 
+                       Content="Select an option to show instructions." 
+                       FontWeight="SemiBold" />
+                <ComboBox x:Name="DamageTypeDropDown"
```
Comment (duffh) [inline src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml:17]: `DamageTypeDropDown` should be hidden unless edit functionality is selected.

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      <StackPanel>
+        <TextBox Text="Select an option." />
+        <ComboBox x:Name="MethodPicker"
+                  IsEditable="False"
+                  IsEnabled="True"
+                  SelectionChanged="OnComboBoxSelectionChanged" />
+        <TextBox x:Name="InstructionLabel"
+                 Text="Select an option to show instructions."
+                 FontWeight="SemiBold" />
+        <ComboBox x:Name="DamageTypeDropDown"
```
Comment (duffh) [inline src/WinUI/ArcGIS.WinUI.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml:19]: `DamageTypeDropDown` should be hidden when edit functionality is not being used.

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    xmlns:esriUI="using:Esri.ArcGISRuntime.UI.Controls">
+  <Grid>
+        <esriUI:MapView x:Name="MyMapView" />
+        <Border Style="{StaticResource BorderStyle}">
+      <StackPanel>
+        <TextBox Text="Select an option." />
+        <ComboBox x:Name="MethodPicker"
+                  IsEditable="False"
+                  IsEnabled="True"
+                  SelectionChanged="OnComboBoxSelectionChanged" />
```
Comment (duffh) [inline src/WinUI/ArcGIS.WinUI.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml:15]: `OnComboBoxSelectionChanged` should un-select any active features.

File: src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+             xmlns:esri="http://schemas.esri.com/arcgis/runtime/2013">
+    <Grid>
+        <esri:MapView x:Name="MyMapView" />
+        <Border Style="{StaticResource BorderStyle}">
+            <StackPanel>
+                <Label Content="Select an option." />
+                <ComboBox x:Name="MethodPicker"
+                          IsEditable="False"
+                          IsEnabled="True"
+                          SelectionChanged="OnComboBoxSelectionChanged" />
```
Comment (duffh) [inline src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml:13]: `OnComboBoxSelectionChanged` should un-select any active features.

File: src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    <Grid>
+        <esriUI:MapView x:Name="MyMapView" />
+        <Border Style="{StaticResource EsriSampleControlPanel}"
+                VerticalOptions="Start">
+            <StackLayout Padding="5"
+                         Spacing="5">
+                <Label Text="Select an option." />
+                <Picker x:Name="MethodPicker"
+                        IsEnabled="True"
+                        SelectedIndexChanged="OnComboBoxSelectionChanged" />
```
Comment (duffh) [inline src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml:15]: `OnComboBoxSelectionChanged` should un-select any active features.

File: src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+             xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:esriUI="clr-namespace:Esri.ArcGISRuntime.Maui;assembly=Esri.ArcGISRuntime.Maui">
+    <Grid>
+        <esriUI:MapView x:Name="MyMapView" />
+        <Border Style="{StaticResource EsriSampleControlPanel}"
+                VerticalOptions="Start">
+            <StackLayout Padding="5"
+                         Spacing="5">
+                <Label Text="Select an option." />
```
Comment (duffh) [inline src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml:12]: Maybe some label like `Select CRUD operation.`

File: src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,443 @@
...
+            MyMapView.GeoViewTapped -= MapView_Tapped_UpdateAttribute;
+            MyMapView.GeoViewTapped -= MapView_Tapped_UpdateGeometry;
+
+            // Hold a variable for the ComboBox's selected value.
+            ComboBox comboBox = (ComboBox)sender;
+
+            if (comboBox.SelectedIndex == 0)
+            {
+                // Update the label.
+                InstructionLabel.Content = "Tap to add a feature.";
```
Comment (duffh) [inline src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs:123]: Maybe `Tap on the GeoView to add a new feature.`

File: src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs [hunk 8]
```diff
@@ -0,0 +1,443 @@
...
+
+                // Update the currently hooked GeoViewTapped event.
+                MyMapView.GeoViewTapped += MapView_Tapped_CreateFeature;
+
+            }
+            else if (comboBox.SelectedIndex == 1)
+            {
+                // Update the label.
+                InstructionLabel.Content = "Tap to delete a feature.";
```
Comment (duffh) [inline src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs:132]: `Tap an existing feature to delete it.`

File: src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,443 @@
...
+
+                // Update the currently hooked GeoViewTapped event.
+                MyMapView.GeoViewTapped += MapView_Tapped_DeleteFeature;
+            }
+            else if (comboBox.SelectedIndex == 2)
+            {
+                // Update the label.
+                InstructionLabel.Content = "Tap to select a feature, then change its damage type.";
```
Comment (duffh) [inline src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs:140]: `Tap an existing feature to edit a property.`

File: src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs [hunk 10]
```diff
@@ -0,0 +1,443 @@
...
+
+                // Update the currently hooked GeoViewTapped event.
+                MyMapView.GeoViewTapped += MapView_Tapped_UpdateAttribute;
+            }
+            else if (comboBox.SelectedIndex == 3)
+            {
+                // Update the label.
+                InstructionLabel.Content = "Tap to select a feature. Tap again to move it.";
```
Comment (duffh) [inline src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs:148]: `Tap an existing feature to select it, tap the GeoView to move it to a new position.`

File: src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml (hunk 6 above)
Comment (williambohrmann3) [reply src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml:12]: I agree with Hamish. I am in favor of a `:` instead of `.` and utilizing bold text it comes to picker instructions. But that's me being nitpicky.

File: src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs (hunk 7 above)
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs:123]: I think `map` might be a little more intuitive than `GeoView`

**Review by williambohrmann3 — CHANGES_REQUESTED** [review]
I had a ton of build errors on WinUI for some reason so couldn't test there.

File: src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml (hunk 6 above)
Comment (amweis226) [reply src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml:12]: I was thinking "Select an edit operation:" as I don't intend to maintain the "CRUD" term

File: src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+                <Picker x:Name="MethodPicker"
+                        IsEnabled="True"
+                        SelectedIndexChanged="OnComboBoxSelectionChanged" />
+                <Label x:Name="InstructionLabel" 
+                       Text="Select an option to show instructions." />
+                <Picker x:Name="DamageTypePicker"
+                        IsVisible="False" />
+                <Button x:Name="DeleteButton"
+                        Clicked="DeleteButton_Click"
+                        IsVisible="False"/>
```
[the remark this answers is not in the document]
Comment (amweis226) [reply src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml:22]: Are you seeing this only in MAUI or in other platforms? This shouldn't be the case

[the remark this answers is not in the document]
Comment (williambohrmann3) [reply src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml:23]: My bad - I meant to comment this for WPF. MAUI behavior is fine.

**Comment by williambohrmann3** [comment]
I believe we should delete the old samples as a part of this PR unless samples design council said otherwise.

**Comment by amweis226** [comment]
> I believe we should delete the old samples as a part of this PR unless samples design council said otherwise.

That's the plan. Wouldn't make sense to have duplicate functionality in the samples.

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+                <TextBlock Padding="5"
+                           Text ="Select an edit operation."
+                           FontWeight="SemiBold" />
+                <ComboBox x:Name="MethodPicker"
+                          HorizontalAlignment="Stretch"
+                          Padding="5"
+                          IsEditable="False"
+                          IsEnabled="True"
+                          SelectionChanged="OnComboBoxSelectionChanged" />
+                <TextBlock x:Name="InstructionLabel"
```
Comment (duffh) [inline src/WinUI/ArcGIS.WinUI.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml:20]: Need to add text wrapping. Currently the `Update geometry` instruction label is cut off before the end of the sentence.

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+                           Padding="5"
+                           Text ="Select an option to show instructions." />
+                <ComboBox x:Name="DamageTypeDropDown"
```
Comment (duffh) [inline src/WinUI/ArcGIS.WinUI.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml:23]: We should include a label above this picker on `WinUI`, `WPF` and `MAUI` that says `Select damage type.` to indicate to the user what they are selecting.

File: src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+        <esri:MapView x:Name="MyMapView" />
+        <Border Style="{DynamicResource BorderStyle}">
+            <StackPanel>
+                <Label Content="Select an edit operation."
+                       FontWeight="SemiBold"/>
+                <ComboBox x:Name="MethodPicker"
+                          IsEditable="False"
+                          IsEnabled="True"
+                          SelectionChanged="OnComboBoxSelectionChanged" />
+                <Label x:Name="InstructionLabel"
```
Comment (duffh) [inline src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml:15]: Need text wrapping here, the `Update geometry` options instruction label is truncated.

File: src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+            <StackPanel>
+                <Label Content="Select an edit operation."
+                       FontWeight="SemiBold"/>
+                <ComboBox x:Name="MethodPicker"
+                          IsEditable="False"
+                          IsEnabled="True"
+                          SelectionChanged="OnComboBoxSelectionChanged" />
+                <Label x:Name="InstructionLabel" 
+                       Content="Select an option to show instructions." />
+                <ComboBox x:Name="DamageTypeDropDown"
```
Comment (duffh) [inline src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml:17]: Add label above combo box `Select damage type.`.

**Review by duffh — CHANGES_REQUESTED** [review]
The screenshots on WPF and WinUI look weirdly stretched. 

Please use the screenshot tool on these platforms, if you need a pointer send me a message and I can show you how to use it.

**Comment by duffh** [comment]
This PR should be targeted at `main` not `v.next`.

File: src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:esriUI="clr-namespace:Esri.ArcGISRuntime.Maui;assembly=Esri.ArcGISRuntime.Maui">
+    <Grid Style="{DynamicResource EsriSampleContainer}">
+        <esriUI:MapView x:Name="MyMapView"
+                        Style="{DynamicResource EsriSampleGeoView}"/>
+        <Border Style="{DynamicResource EsriSampleControlPanel}"
+                VerticalOptions="Start">
+            <StackLayout Padding="5"
+                         Spacing="5">
+                <Label Text="Select an edit operation:" />
```
[the remark this answers is not in the document]
Comment (amweis226) [reply src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml:13]: Per ArcGIS Pro spec, "Edit" is an umbrella term for managing attributes within a feature (attribute) table. This includes creation and deletion. 

That being said, `Select a feature management operation` is more consistent with the terminology of the sample itself.

File: src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/readme.md [hunk 17]
```diff
@@ -0,0 +1,41 @@
+# Manage features
+
+Manage a feature layer's features in four distinct ways.
+
+![Image of managing features](ManageFeatures.jpg)
+
+## Use case
+
+An end-user performing a survey may want to edit features on the map in various ways during the course of their work.
```
[the remark this answers is not in the document]
Comment (amweis226) [reply src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/readme.md:9]: Per ArcGIS Pro spec, "edit" is an umbrella term for managing attributes within a feature (attribute) table. This includes creation and deletion. I didn't want to make it too wordy and it's technically correct as-is. However, I could change it to say something like `create and delete features or update attributes and geometries` if desired.

**Comment by williambohrmann3** [comment]
I don't think we need to specify a width for MAUI pickers here. When `Picker` is in a `Grid`, it fills out horizontally on all platforms excluding Windows. This [issue](https://github.com/dotnet/maui/issues/6391) was reported in the MAUI repo.

File: src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml (hunk 16 above)
[the remark this answers is not in the document]
Comment (williambohrmann3) [reply src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml:13]: Could we just shorten it to `Select an operation:`?

File: src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+        <Border Style="{DynamicResource BorderStyle}">
+            <StackPanel>
+                <Label Content="Select a feature management operation."
+                       FontWeight="SemiBold"/>
+                <ComboBox x:Name="MethodPicker"
+                          IsEditable="False"
+                          IsEnabled="True"
+                          SelectionChanged="OnComboBoxSelectionChanged" />
+                <Label x:Name="InstructionLabel" 
+                       Content="Select an option to show instructions." />
```
Comment (williambohrmann3) [inline src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml:16]: To avoid redundancy, I think this could be an empty string upon starting sample. We are already telling user to make a selection.

File: src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs [hunk 19]
```diff
@@ -0,0 +1,439 @@
...
+                    // Update the label.
+                    InstructionLabel.Text = "Tap an existing feature to edit its attribute.";
+
+                    // Update the currently hooked GeoViewTapped event.
+                    MyMapView.GeoViewTapped += MapView_Tapped_UpdateAttribute;
+                }
+                else if (picker.SelectedIndex == 3)
+                {
+                    // Update the label.
+                    InstructionLabel.Text = "Tap an existing feature to select it, tap the map to move it to a new position.";
```
Comment (williambohrmann3) [inline src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs:152]: Maybe something like "Tap an existing feature to select it and again to move it." to make it less wordy?

File: src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs [hunk 20]
```diff
@@ -0,0 +1,439 @@
...
+
+                // Update instruction label.
+                InstructionLabel.Text = "Select damage type:";
+
+                // Update the UI for the selection.
+                UpdateUIForSelectedFeature();
+            }
+            catch (Exception ex)
+            {
+                await Application.Current.MainPage.DisplayAlert("Error selecting feature.", ex.ToString(), "OK");
```
Comment (williambohrmann3) [inline src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs:311]: Minor nitpick - alert titles don't need periods

File: src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+        <esri:MapView x:Name="MyMapView" />
+        <Border Style="{DynamicResource BorderStyle}">
+            <StackPanel>
+                <Label Content="Select a feature management operation."
+                       FontWeight="SemiBold" />
+                <ComboBox x:Name="MethodPicker"
+                          IsEditable="False"
+                          IsEnabled="True"
+                          SelectionChanged="OnComboBoxSelectionChanged" />
+                <Label x:Name="InstructionLabel" />
```
Comment (williambohrmann3) [inline src/WPF/WPF.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml:15]: This needs text wrapping - update geometry instruction is cut off. Same issue in WinUI.

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml [hunk 22]
```diff
@@ -0,0 +1,28 @@
+<UserControl 
+    x:Class="ArcGIS.WinUI.Samples.ManageFeatures.ManageFeatures"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:esri="using:Esri.ArcGISRuntime"
+    xmlns:esriUI="using:Esri.ArcGISRuntime.UI.Controls">
+    <Grid>
+        <esriUI:MapView x:Name="MyMapView" />
+        <Border Style="{StaticResource BorderStyle}">
```
Comment (williambohrmann3) [inline src/WinUI/ArcGIS.WinUI.Viewer/Samples/Data/ManageFeatures/ManageFeatures.xaml:9]: Totally your choice - here and in WPF, you could try shrinking the border width a bit.

**Review by williambohrmann3 — APPROVED** [review]
Definitely ensure the instruction label has text wrapping in WinUI and WPF. Also include sample in featured category. Otherwise, this looks great.

**Comment by amweis226** [comment]
> Definitely ensure the instruction label has text wrapping in WinUI and WPF. Also include sample in featured category. Otherwise, this looks great.

@duffh, your opinion on adding this to favorites?

File: src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs [hunk 23]
```diff
@@ -0,0 +1,439 @@
...
+        "Data",
+        "Create or delete features, update attributes or geometry.",
+        "")]
+    [ArcGIS.Samples.Shared.Attributes.OfflineData()]
+    public partial class ManageFeatures
+    {
+        // URL to the feature service.
+        private const string FeatureServiceUrl = "https://sampleserver6.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer/0";
+
+        // Hold a reference to the feature layer
```
Comment (duffh) [inline src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs:29]: Missing `.` at the end of the comment.

**Review by duffh — CHANGES_REQUESTED** [review]
Looks good just a few minor things.

File: src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs (hunk 23 above)
Comment (amweis226) [reply src/MAUI/Maui.Samples/Samples/Data/ManageFeatures/ManageFeatures.xaml.cs:29]: Good catch

**Review by duffh — APPROVED** [review]
Approved pending fixed merge conflicts. 

Check that the GitHub actions are passing also before merging.
